extension/keysocket-deezer.js

```diff
@@ -1,38 +1,23 @@
 function onKeyPress(key) {
-    if (document.getElementById('player_control_play')) { // old deezer style
+	if(document.querySelector('#player_control_play')) { // old deezer style
 		if(key === NEXT) {
-			var nextButton = document.getElementById('player_control_next');
-			simulateClick(nextButton);
+			simulateClick(document.querySelector('#player_control_next'));
 		} else if(key === PLAY) {
-			var isPlaying = document.getElementById('player_control_play').style.display === 'none';
-			var playPauseButton = null;
-			if(isPlaying) {
-			  playPauseButton = document.getElementById('player_control_pause');
-			} else {
-			  playPauseButton = document.getElementById('player_control_play');
-			}
-			simulateClick(playPauseButton);
+			simulateClick(document.querySelector('#player_control_play').style.display === 'none' ? document.querySelector('#player_control_pause') : document.querySelector('#player_control_play'));
```

Comment:
I think I see the direction you're wanting to go. This will make the code a lot bigger in terms of what you're looking at visually when you open a file. I don't really think this will simplify the code.

As for the case switch, I like the idea, but I don't think it'll be helpful in the goal of simplifying things. I don't think case switch will work any better than if/else for the majority of the code here. 

I'll write up a mock to try out the example you posted and I'll put it here in the comments.

Another file I have concerns with is the YouTube socket. The changes I posted work, but looking at the code isn't obvious as to what's going on. I'll probably have to rewrite that one.